KSpaceSolver/KSpaceFirstOrderSolver.cpp

```diff
@@ -472,6 +472,14 @@ size_t KSpaceFirstOrderSolver::getMemoryUsage() const
 
     return pmc.PeakWorkingSetSize >> 20;
   #endif
+
+  // macOS build
+  #ifdef __APPLE__
+    struct rusage memUsage;
+    getrusage(RUSAGE_SELF, &memUsage);
+
+    return memUsage.ru_maxrss >> 10;
+  #endif
```

Comment:
This unifies this guard with the Linux guard above, allowing us to remove it.
```suggestion
// Unix-like systems (Linux and macOS)
#if defined(__linux__) || defined(__APPLE__)
  struct rusage memUsage;
  getrusage(RUSAGE_SELF, &memUsage);
  return memUsage.ru_maxrss >> 10;
#endif
```